eng/liveBuilds.targets

```diff
@@ -6,28 +6,73 @@
+  <!--
+    If this is running and the output RID is not the same as the targeted RID, resolve live assets
+    for the targeted RID, if available. This is used to gather asset metadata for the platform
+    manifest. In CI (multi-machine lab) builds, CoreCLR and Libraries artifacts are all downloaded
+    onto the current machine from all platforms for the Installer portion of the build.
 
+    Higher priority than override paths: in official builds, the overrides are passed to the build,
+    but we need to point to the AllArtifacts locations when building the platform manifest.
+  -->
+  <PropertyGroup Condition="
+    '$(RuntimeIdentifier)' != '$(OutputRid)' and
+    '$(AllArtifactsDownloadPath)' != ''">
+    <!-- Convert the OS component in the RID into names that match the job IDs. -->
+    <ArtifactPlatform>$(RuntimeIdentifier)</ArtifactPlatform>
+    <ArtifactPlatform>$(ArtifactPlatform.Replace('win-', 'Windows_NT-'))</ArtifactPlatform>
+    <ArtifactPlatform>$(ArtifactPlatform.Replace('unix-', 'Unix-'))</ArtifactPlatform>
+    <ArtifactPlatform>$(ArtifactPlatform.Replace('linux-', 'Linux-'))</ArtifactPlatform>
+    <ArtifactPlatform>$(ArtifactPlatform.Replace('osx-', 'OSX-'))</ArtifactPlatform>
+    <ArtifactPlatform>$(ArtifactPlatform.Replace('freebsd-', 'FreeBSD-'))</ArtifactPlatform>
+    <ArtifactPlatform>$(ArtifactPlatform.Replace('netbsd-', 'NetBSD-'))</ArtifactPlatform>
+    <!-- Artifact name uses '_' rather than '-'. -->
+    <ArtifactPlatform>$(ArtifactPlatform.Replace('-', '_'))</ArtifactPlatform>
+
+    <CoreCLRArtifactsPath>$([MSBuild]::NormalizeDirectory('$(AllArtifactsDownloadPath)', 'CoreCLRProduct_$(ArtifactPlatform)_$(CoreCLRConfiguration)'))</CoreCLRArtifactsPath>
+    <LibrariesArtifactsPath>$([MSBuild]::NormalizeDirectory('$(AllArtifactsDownloadPath)', 'libraries_bin_$(ArtifactPlatform)_$(LibrariesConfiguration)'))</LibrariesArtifactsPath>
+
+    <!--
+      Use '*', calculating these in a static context based on RID is hard. Note that putting '*'
+      through a dir/path normalization function makes it not work, so avoid that.
+    -->
+    <LibrariesOSGroupConfigurationArchitecture>*</LibrariesOSGroupConfigurationArchitecture>
+  </PropertyGroup>
+
+  <!-- Set up default live asset paths if no overrides provided. -->
+  <PropertyGroup>
+    <CoreCLRArtifactsPath Condition="'$(CoreCLRArtifactsPath)' == ''">$([MSBuild]::NormalizeDirectory('$(RepoRoot)', 'artifacts', 'bin', 'coreclr', '$(CoreCLROSGroup).$(TargetArchitecture).$(CoreCLRConfiguration)'))</CoreCLRArtifactsPath>
     <LibrariesArtifactsPath Condition="'$(LibrariesArtifactsPath)' == ''">$([MSBuild]::NormalizeDirectory('$(RepoRoot)', 'artifacts'))</LibrariesArtifactsPath>
+    <LibrariesAllConfigurationsArtifactsPath Condition="'$(LibrariesAllConfigurationsArtifactsPath)' == ''">$([MSBuild]::NormalizeDirectory('$(RepoRoot)', 'artifacts'))</LibrariesAllConfigurationsArtifactsPath>
```

Comment:
Should we also normalize the all configurations path if it is passed down?

Reply:
I didn't need to split this prop into `LibrariesAllConfigurationsOverridePath` and `LibrariesAllConfigurationsArtifactsPath`, because we don't have multiple AllConfigurations runs and the platform manifest doesn't need AllConfigurations artifacts anyway.

For readability though, doing this here too makes sense. (Removes the need to ask "why are there Override props for these two artifact paths and not this other one?") Filed https://github.com/dotnet/runtime/issues/1615 for that. (Avoids respinning CI.)